src/Symfony/Component/DependencyInjection/Tests/Compiler/ValidateConstructorArgumentsPassTest.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Tests\Compiler;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\DependencyInjection\Compiler\ValidateConstructorArgumentsPass;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\Validator\Constraints\Ip;
+use Symfony\Component\Validator\Exception\ValidationFailedException;
+
+class ValidateConstructorArgumentsPassTest extends TestCase
+{
+    public function testValidationSuccess()
+    {
+        $container = new ContainerBuilder();
+        $definition = $container->register('service', \stdClass::class);
+        $definition
+            ->setArguments([
+                '$int' => 1,
+                '$array' => [1, 2, 3],
+                '$email' => 'test@email.com',
+                '$datetime' => '2020-12-31 23:59:59',
+                '$ipAddresses' => ['8.8.4.4', '8.8.8.8'],
+                '$noConstraints' => 'no constraints for this argument',
```

Comment:
Is this used?

Reply:
@OskarStark, sorry, didn't understand what do you mean